### PR TITLE
Adding podAnnotations to StatsD deployment template

### LINKED ISSUE #25446: Helm Chart: Allow adding annotations to statsd deployment
### Description

Helm Chart [does not allow adding annotations](https://github.com/apache/airflow/blob/40eefd84797f5085e6c3fef6cbd6f713ceb3c3d8/chart/templates/statsd/statsd-deployment.yaml#L60-L63) to StatsD deployment. We should add it.

### Use case/motivation

In our Kubernetes cluster we need to set annotations on deployments that should be scraped by Prometheus. Having an exporter that does not get scraped defeats the purpose :slightly_smiling_face:

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by joshuaghezzi**
Adding PodAnnotation to StatsD deployment template.
Closes: https://github.com/apache/airflow/issues/25446

### CODE REVIEW

**Comment by sbrandtb** [comment]
Please, next time you are working on an issue that is assigned to someone else, please notify the issue. I do not want to sound rude, but it is simple fact that I just wasted time on this because I implemented it as well. Time that could have been spent more wisely.

**Comment by potiuk** [comment]
> Please, next time you are working on an issue that is assigned to someone else, please notify the issue. I do not want to sound rude, but it is simple fact that I just wasted time on this because I implemented it as well. Time that could have been spent more wisely.

I think the problem was .... you were not assigned @sbrandtb :). It was likely an omission :) when you read all the conversation it looked like you wanted to work on it , but yeah - it's our fault we have not assigned you.

BTW. I think it's quite expected that there is some duplication of work. There is also a value in doing stuff (learning). Merging is not the ultimate and only value, the process of making PR. getting reviews and feedback is valuable as well. There is no sane way to make sure that there is no duplication of work, but when you realise this is not the only value, you get more acceptance of doing stuff in parallel by different people. Code is often liability and if you get others contribute the code you came with as an idea, this is actually cool opportunity to work on it together.

Since this PR is alrady out - in the spirit of what I wrote above - I'd encourage you @sbrandtb to turn into reviewer here and make your comments and help with leading that to completion. There is even bigger value in being reviewer when such code is contributed - you learn how to communicate your review comments, you learn how to look at the change with other perspective and you will likely notice things that you wouldn't have noticed when you wrote it. And a good learning experience too.

At the end "Community over code" is what matters most (this is the ASF motto).

**Comment by sbrandtb** [comment]
@potiuk As mentioned in the issue: No offense. And MR looks good to me, but I do not have the power to +1

File: chart/templates/statsd/statsd-deployment.yaml [hunk 1]
```diff
@@ -57,9 +57,17 @@ spec:
 {{- with .Values.labels }}
 {{ toYaml . | indent 8 }}
 {{- end }}
-{{- if .Values.statsd.extraMappings }}
+{{- if or .Values.statsd.extraMappings .Values.airflowPodAnnotations .Values.statsd.podAnnotations }}
```
Comment (jedcunningham) [inline chart/templates/statsd/statsd-deployment.yaml]:
```suggestion
{{- if or .Values.statsd.extraMappings .Values.statsd.podAnnotations }}
```

Let's not include the `airflowPodAnnotations`. This isn't an Airflow pod.

File: chart/templates/statsd/statsd-deployment.yaml [hunk 2]
```diff
@@ -57,9 +57,17 @@ spec:
...
       annotations:
+        {{- if .Values.statsd.extraMappings }}
         checksum/statsd-config: {{ include (print $.Template.BasePath "/configmaps/statsd-configmap.yaml") . | sha256sum }}
+        {{- end }}
+        {{- if .Values.airflowPodAnnotations }}
+        {{- toYaml .Values.airflowPodAnnotations | nindent 8 }}
+        {{- end }}
```
Comment (jedcunningham) [inline chart/templates/statsd/statsd-deployment.yaml]:
```suggestion
```

**Review by jedcunningham — APPROVED** [review]

**Pull request merged**